addons/sale_management/models/sale_order_line.py

```diff
@@ -33,9 +33,13 @@ def _use_template_name(self):
     def _compute_price_unit(self):
         # Avoid recomputing the price with pricelist rules, use the initial price
         # used in the optional product line.
-        optional_product_lines = self.filtered('sale_order_option_ids')
+        optional_product_lines = self._lines_to_avoid_recompute_price()
         super(SaleOrderLine, self - optional_product_lines)._compute_price_unit()
 
+    def _lines_to_avoid_recompute_price(self):
```

Comment:
Not sure about this one, feel free to keep your version ^^

```suggestion
    def _lines_without_price_recomputation(self):
```